warehouse/legacy/api/json.py

```diff
@@ -217,3 +217,120 @@ def json_release_slash(release, request):
         ),
         headers=_CORS_HEADERS,
     )
+
+
+@view_config(
+    route_name="legacy.api.json.latest",
+    context=Project,
+    renderer="json",
+    decorator=_CACHE_DECORATOR,
+)
+def json_latest(project, request):
```

Comment:
As written, this function (and those analogous to it) make two database calls. The first is within the implementation of `project.latest_version`, and the second is a few lines below, when `release` is defined.

Would it be better from a performance or other perspective to figure out how to refactor this down to a single database access? (I'm not entirely sure if it's possible, though, given that the web view logic needs something different from the JSON view logic.)

Reply:
IMHO, 2 DB calls is still early optimization, if we're not on the critical path.

Reply:
IMHO, trying to remove 1 DB call if it has readability implications and we're not on the critical path is early optimization.